neuralprophet/conformal_prediction.py

```diff
@@ -0,0 +1,109 @@
+import matplotlib
+import pandas as pd
+
+from neuralprophet.plot_forecast_matplotlib import plot_nonconformity_scores
+from neuralprophet.plot_forecast_plotly import plot_nonconformity_scores as plot_nonconformity_scores_plotly
+
+
+def conformalize(df_cal, alpha, method, quantiles, plotting_backend):
+    """Apply a given conformal prediction technique to get the uncertainty prediction intervals (or q-hats).
+
+    Parameters
+    ----------
+        df_cal : pd.DataFrame
+            calibration dataframe
+        alpha : float
+            user-specified significance level of the prediction interval
+        method : str
+            name of conformal prediction technique used
+
+            Options
+                * (default) ``naive``: Naive or Absolute Residual
+                * ``cqr``: Conformalized Quantile Regression
+
+        quantiles : list
+            list of quantiles for quantile regression uncertainty estimate
```

Comment:
Those conformal prediction rely on the quantile regression as it reuses those parameters or are those separate ones? Just asking as I'm trying to better understand how those to methods are connected (or not)?

Reply:
Right now there are two available conformal prediction methods: **naive** and **CQR**. Naive does conformal prediction independently from the QR while CQR applies conformal prediction on the QR itself rather than the point prediction.